### DISCUSSION TITLE
Modern Search v4 - October 2021

### DISCUSSION THREAD

**Discussion by FranckyC** [microsoft-search/pnp-modern-search#1458] [Announcements]
# PnP Modern Search v4

**This solution includes now only one `.sppkg` file:**
- `pnp-modern-search-parts-v4.sppkg`. Mandatory.

**Getting started documentation here:** [https://microsoft-search.github.io/pnp-modern-search/](https://microsoft-search.github.io/pnp-modern-search/)

[Feedback and comments](https://github.com/microsoft-search/pnp-modern-search/issues) are welcomed! 

## Important - Important - Important
- IE11 support removed
- There's been an issue where `onclick`, `onload` etc. has not been sanitized correctly. This is now fixed so script (injection) in templated should now be fully blocked.
   - Custom templates with fallback/error templates on images are affected, and will require updates!
- Query variable handling now aligns with the classic implementation, and how Microsoft Search will handle them. A missing variable needs to have the `{?...}` notation for the part to be removed from the KQL. It will not be replaced with `*`.

## General

- [Documentation] Clarify how to add managed properties to the selected properties in search results.

### New features

- #1232 - Adds ability to sort columns with Handlebars expressions on Details List
- #1317 - Pick language from page translation over page context if available
- #1272 - Update `getAttachments` Handlebars helper code to include index
- #1402 - Added accessibility

### Fixes
- #1352 - Wrong Handlebars context for the `groupBy` helper. 
- #1188 - Updated the thumbnail and preview logic to not load thumbnail for non-SPO domains and interactive preview for different domain than Web Part.
- #1329 - Cards template with odd numbers
- #1277 - Fix preview for folders and fix icon for OneNote's
- #1236 - Double encoding for queryText in Search Box
- #1204 - Fix/Normalization for Taxonomy Refiner Localization
- #1178 - Fixed warning messages due to multiple file type icons load in the template service 
- #1311 - Use correct UTZ date as bases
- #1178 - Optimize UI fabric styling and console warnings
- #1183 - Fixed some nl translations
- #1231 - Fix parsing of `{?QueryString.Parameter}` query variable notation
- #1252 - Allow pasting comma separated list of managed properties into selected properties
- #1191 - CSS fix for the persona template
- #1189 - Pick language from page translation over page context if available
- #1329 - Fixed cards template with odd numbers
- #1267 - Search Filters & Search Results break when lazy loading from header link
- #1348 - Fix preview and icon for site results in list template
- #1328 - Split on | for User multivalue tokens
- #1390 - Add `pnp-img` to be used instead of `img` tag and fix sanitization of markup attributes
- #1392 - Fixed the 'default' helper
- #1339 - Fix for % in query term
- #1046 - UX changes to error message handling

### Fixes
- #1225 - Sticky "Apply | Clear" buttons in ComboBox
- #1222 - Fix touch scroll on filter panel
- #1323 - Filter combo control stops to "search as you type" after closing dropdown once with Esc-key or mouse.
- #1287 - Filters Modified Date Interval Wording
- #1144 - Filter panel appears when clearing search terms
- #1165 - Fix combo box selection issue
- #1185 - Allow accented characters in filter web part's filter box
- #1267 - Search Filters & Search Results break when lazy loading from header link
- #1287 - Filters Modified Date Interval Wording
- #1444 - Web Part does not re-render after a pushState when query string parameter is used in tokens
- #1402 - Accessibility compatibility 

### New Features
- #1282 - Adds ability to sort columns with handlebar expressions on Details List
- #1402 - Added accessibility
- #1364 - Add support for `person`  and `list` entity on the graph provider.
- #1428 - Added polish translations
- #1367 - Added filter deep links feature
- #1450 - Improved Swedish translations  

### New features
- #1402 - Added accessibility

### Fixes
- #1339 - Fix for % in query term

## Release PR link
#1426 All commits

## Key contributors

- @ellinge
- @wobba
- @juliemturner
- @sympmarc 
- @mrpullen
- @jhholm 
- @svermaak
- @Laetterman
- @CreativeAcer 
- @dennisvanniel 
- @FranckyC 
- @MakingThingsWorkToronto 
- @dennisvanniel
- @jimmywim 
- @fdafadf 
- @sherpas-par

<hr /><em>This discussion was created from the release <a href='https://github.com/microsoft-search/pnp-modern-search/releases/tag/4.4.1'>Modern Search v4 - October 2021</a>.</em>

**Comment by sumitdeyonline** [comment]
We have created two items in the search verticle web part(Everything and People), can we use two separate Result Layouts in the search result web part? we want to use the "Simple List" layout for the "Everything" tab and the "People" layout for the People tab. 

![image](https://user-images.githubusercontent.com/23446009/141009238-4e658e2d-481a-49ff-9cd7-fd4111856616.png)
**Everything Result ===>**

![image](https://user-images.githubusercontent.com/23446009/141009579-f87541c5-6d07-4bd7-8c9f-bc7065e0240c.png)
**People Result ===>**

![image](https://user-images.githubusercontent.com/23446009/141009926-505f1097-198f-4924-af57-b40040b87ec2.png)

**Reply by sympmarc** [reply]
Best way to do this in my experience is to either:
* Add multiple Search Result Web Parts, each connected to a vertical and/or
* Create separate pages for each vertical

**Comment by mariaarcena** [comment]
@FranckyC  Hello! Is is the Microsoft Search data source can be used in production now?

**Reply by FranckyC** [reply]
@mariaarcena the endpoint still points to `beta` so I would say no. We will update this for the next release, thanks for the reminder ;)

**Reply by mariaarcena** [reply]
Oh! So it was planned to use the non-beta after all? got it! thank you so much! ☺️  (unless you think it should be easy for me to change the endpoint within source code again but I'm afraid that might break the solution if I were to do that since I am new to sharepoint framework) 😅 but yes, I would wait for release then! 😊

**Reply by FranckyC** [reply]
The Microsoft Graph search API is available in v1.0 so yes it can be used in production https://docs.microsoft.com/en-us/graph/api/resources/search-api-overview?view=graph-rest-1.0. However I think filters only work with `beta` right now.

**Reply by wobba** [reply]
We'll update the docs to reflect what is beta and not. And beta is fairly stable with very little change - but it can change.

**Comment by Selamyi** [comment]
I have downloaded and start using Modern Search v4 web part. I have a custom column called mattername. i would like to use this custom column in the search result web part.  But the results are not showing for this custom column.

**Reply by jimmywim** [reply]
Make sure you've selected mattername from the Selected Properties dropdown box.

**Reply by FranckyC** [reply]
Or add it manually and press 'Enter' in the selected properties dropdown text field

**Reply by Selamyi** [reply]
Hello immywim. Thanks for replay.
yes i have selected mattername from the Selected Properties dropdown box

**Reply by jimmywim** [reply]
Does the mattername property show any data when you use Debug view?

https://microsoft-search.github.io/pnp-modern-search/usage/search-results/layouts/#debug

**Reply by Selamyi** [reply]
@FranckyC  I have done it but no luck

**Reply by Selamyi** [reply]
@jimmywim its showing null

**Reply by jimmywim** [reply]
Have you configured a Search Managed Property for your column?

You may need to map mattername to an existing RefinableString slot, or create a new Managed Property and map it to mattername (although you won't be able to refine or sort, so mapping to existing RefinableString properties is the recommended approach).

https://docs.microsoft.com/en-us/sharepoint/manage-search-schema#map-a-crawled-property-to-a-managed-property

**Reply by Selamyi** [reply]
And one more question. how can i make it to search only on the specific site. its now searching tenant wide

**Reply by Selamyi** [reply]
@FranckyC @jimmywim thanks it works when i just remove everything and do fresh install.

**Comment by chriscundy** [comment]
Is it possible to filter by the "Created" column in the search results, similar to in a list? For example newer to older.

![image](https://user-images.githubusercontent.com/543811/147754084-cc4590df-3c02-4858-b631-0004b451e79e.png)

**Reply by wobba** [reply]
Sorting using the details list will sort on-page items only - not issue a new query with sorting. v4 don't have the sorting drop-down like v3 did.
